File: Sources/custom_adapter/TopOnAdapter/Objc/6.5.x/Base/AlxTopOnInitAdapter.m
//
//  AlxTopOnInitAdapter.m
//  AlxAdsOCDemo
//

#import "AlxTopOnInitAdapter.h"
#import <AdSupport/AdSupport.h>

@implementation AlxTopOnInitAdapter

/// 初始化 Ad SDK
/// Initialize the Ad SDK.
/// - Parameter adInitArgument: 包含后台配置信息的参数对象 / Parameter object containing server configuration info.
- (void)initWithInitArgument:(ATAdInitArgument *)adInitArgument {
    NSLog(@"AlxTopOnInitAdapter: initWithInitArgument");
    
    // 从 adInitArgument 对象中获取后台配置的信息
    // Retrieve server-configured parameters from the adInitArgument object
    NSString *appid = adInitArgument.serverContentDic[@"appid"];
    NSString *sid = adInitArgument.serverContentDic[@"sid"];
    NSString *token = adInitArgument.serverContentDic[@"token"];
    NSString *debug = adInitArgument.serverContentDic[@"isdebug"];
    
    NSLog(@"AlxTopOnInitAdapter: appid = %@, sid = %@, token = %@, debug = %@", appid, sid, token, debug);
    
    // 参数检查 / Parameter validation
    if (!appid || !sid || !token) {
        NSString *errorStr = @"initialize alx params: appid or sid or token is empty";
        NSLog(@"AlxTopOnInitAdapter: error: %@", errorStr);
        NSError *error = [NSError errorWithDomain:@"AlxTopOnAdapter" code:-100 userInfo:@{NSLocalizedDescriptionKey: errorStr}];
        // 通知 TopOn SDK 初始化失败 / Notify TopOn SDK of initialization failure
        [self notificationNetworkInitFail:error];
        return;
    }
    
    dispatch_async(dispatch_get_main_queue(), ^{
        // 初始化 Alx SDK / Initialize Alx SDK
        [AlxSdk initializeSDKWithToken:token sid:sid appId:appid];
        
        // 设置调试模式 / Set debug mode
        if (debug.length > 0) {
            if ([debug.lowercaseString isEqualToString:@"true"]) {
                [AlxSdk setDebug:YES];
            } else if ([debug.lowercaseString isEqualToString:@"false"]) {
                [AlxSdk setDebug:NO];
            }
        }
        
        // 设置隐私合规 / Configure privacy compliance
        NSInteger gdprFlag = [[NSUserDefaults standardUserDefaults] integerForKey:@"IABTCF_gdprApplies"];
        NSString *gdprConsent = [[NSUserDefaults standardUserDefaults] stringForKey:@"IABTCF_TCString"];
        
        if (gdprFlag == 1) {
            [AlxSdk setGDPRConsent:YES];
        } else {
            [AlxSdk setGDPRConsent:NO];
        }
        [AlxSdk setGDPRConsentMessage:gdprConsent ?: @""];
        
        // 记录 SDK 信息 / Record SDK info
        NSDictionary *data = @{
            @"sdk_name": @"TopOn",
            @"sdk_version": [[ATAPI sharedInstance] version] ?: @"",
            @"adapter_version": [self adapterVersion] ?: @""
        };
        [AlxSdk addExtraParametersWithKey:@"alx_adapter" value:data];
        
        NSLog(@"AlxTopOnInitAdapter: init success");
        // ⚠️ 注意：通知 TopOn SDK 初始化成功
        // ⚠️ Note: Notify TopOn SDK of successful initialization
        [self notificationNetworkInitSuccess];
    });
}

/// 返回广告平台 SDK 的版本号
/// Returns the version number of the ad platform SDK.
- (nullable NSString *)sdkVersion {
    return AlxSdk.getSDKVersion;
}

/// 返回适配器版本号
/// Returns the adapter version number.
- (nullable NSString *)adapterVersion {
    return @"1.3.0";
}


@end

File: Sources/custom_adapter/TopOnAdapter/Swift/6.5.x/Base/AlxTopOnInitAdapter.swift
//
//  AlxTopOnInitAdapter.swift
//  AlxAdsDemo
//
//  直接翻译 OC 代码
//  Translated directly from OC code
//

import Foundation
import AnyThinkSDK
import AlxAds

@objc(AlxTopOnInitAdapter)
public class AlxTopOnInitAdapter: ATBaseInitAdapter {
    private static let TAG = "AlxTopOnBaseInitAdapter"
    
    @objc public override func initWith(_ adInitArgument: ATAdInitArgument) {
        NSLog("AlxTopOnInitAdapter: initWith")
        NSLog("%@: alx-sdk-version:%@", AlxTopOnInitAdapter.TAG, AlxSdk.getSDKVersion())
        NSLog("%@: topon-sdk-version:%@", AlxTopOnInitAdapter.TAG, ATAPI.sharedInstance().version())
        NSLog("%@: topon-adapter-version:%@", AlxTopOnInitAdapter.TAG, "\(self.adapterVersion() ?? "")")
        
        guard let appid = adInitArgument.serverContentDic["appid"] as? String,
              let sid = adInitArgument.serverContentDic["sid"] as? String,
              let token = adInitArgument.serverContentDic["token"] as? String else {
            let errorStr = "initialize alx params: appid or sid or token is empty"
            NSLog("AlxTopOnInitAdapter: error: \(errorStr)")
            let error = NSError(domain: "AlxTopOnAdapter", 
                              code: -100, 
                              userInfo: [NSLocalizedDescriptionKey: errorStr])
            notificationNetworkInitFail(error)
            return
        }
        
        let debug = adInitArgument.serverContentDic["isdebug"] as? String
        
        NSLog("AlxTopOnInitAdapter: appid = \(appid), sid = \(sid), token = \(token), debug = \(debug ?? "nil")")
        
        DispatchQueue.main.async {
            AlxSdk.initializeSDK(token: token, sid: sid, appId: appid)
            
            if let debug = debug, debug.count > 0 {
                if debug.lowercased() == "true" {
                    AlxSdk.setDebug(true)
                } else if debug.lowercased() == "false" {
                    AlxSdk.setDebug(false)
                }
            }
            
            // 读取 GDPR 合规标志 / Read GDPR compliance flags
            let gdprFlag = UserDefaults.standard.integer(forKey: "IABTCF_gdprApplies")
            let gdprConsent = UserDefaults.standard.string(forKey: "IABTCF_TCString")
            
            if gdprFlag == 1 {
                AlxSdk.setGDPRConsent(true)
            } else {
                AlxSdk.setGDPRConsent(false)
            }
            AlxSdk.setGDPRConsentMessage(gdprConsent ?? "")
            
            // 上报适配器信息 / Report adapter info
            let data: [String: Any] = [
                "sdk_name": "TopOn",
                "sdk_version": ATAPI.sharedInstance().version(),
                "adapter_version": AlxTopOnInitAdapter.adapterVersion() ?? ""
            ]
            AlxSdk.addExtraParameters(key: "alx_adapter", value: data)
            
            NSLog("AlxTopOnInitAdapter: init success")
            self.notificationNetworkInitSuccess()
        }
    }
    
    /// 返回广告平台 SDK 的版本号
    /// Returns the version number of the ad platform SDK.
    @objc public func sdkVersion() -> String? {
        return AlxSdk.getSDKVersion()
    }
    
    /// 返回适配器版本号
    /// Returns the adapter version number.
    @objc public func adapterVersion() -> String? {
        return "1.3.0"
    }
}
